app/models/patient.rb

```diff
@@ -224,8 +224,9 @@ def anonymized_data
      gender: gender}
   end
 
-  def discard_data
+  def discard_data(deleted_reason: "unknown")
```

Comment:
I'm adding a default `deleted_reason` to an explicit `unknown`, after seeing the possible values in `DELETED_REASONS`. Before this, the default was unintentionally `nil`. Grepping the code, it shouldn't affect any reporting query, but let me know if you prefer default should remain `nil`.

I have also ignored migrating the existing `nil` value columns to `unknown`s for now.

Reply:
Let's make this a required parameter so a user of this class will need to supply a reason and we don't have to overload `unknown`.